coderd/database/dbauthz/dbauthz_test.go

```diff
@@ -1470,6 +1470,22 @@ func (s *MethodTestSuite) TestWorkspace() {
 		// No asserts here because SQLFilter.
 		check.Args(database.GetWorkspacesParams{}, emptyPreparedAuthorized{}).Asserts()
 	}))
+	s.Run("GetWorkspacesAndAgentsByOwnerID", s.Subtest(func(db database.Store, check *expects) {
+		ws := dbgen.Workspace(s.T(), db, database.WorkspaceTable{})
+		build := dbgen.WorkspaceBuild(s.T(), db, database.WorkspaceBuild{WorkspaceID: ws.ID, JobID: uuid.New()})
+		_ = dbgen.ProvisionerJob(s.T(), db, nil, database.ProvisionerJob{ID: build.JobID, Type: database.ProvisionerJobTypeWorkspaceBuild})
+		res := dbgen.WorkspaceResource(s.T(), db, database.WorkspaceResource{JobID: build.JobID})
+		_ = dbgen.WorkspaceAgent(s.T(), db, database.WorkspaceAgent{ResourceID: res.ID})
+		check.Args(ws.OwnerID).Asserts()
```

Comment:
Doesn't this need to assert Read on the workspace?

Reply:
The SQL filtering doesn't call `Authorize`, it calls `CompileToSQL`. My understanding is that's why the other queries that use the sql filter have empty asserts too:

```
	s.Run("GetWorkspaces", s.Subtest(func(db database.Store, check *expects) {
		_ = dbgen.Workspace(s.T(), db, database.WorkspaceTable{})
		_ = dbgen.Workspace(s.T(), db, database.WorkspaceTable{})
		// No asserts here because SQLFilter.
		check.Args(database.GetWorkspacesParams{}).Asserts()
	}))
	s.Run("GetAuthorizedWorkspaces", s.Subtest(func(db database.Store, check *expects) {
		_ = dbgen.Workspace(s.T(), db, database.WorkspaceTable{})
		_ = dbgen.Workspace(s.T(), db, database.WorkspaceTable{})
		// No asserts here because SQLFilter.
		check.Args(database.GetWorkspacesParams{}, emptyPreparedAuthorized{}).Asserts()
	}))
```

I forgot to copy the comment, so I'll add it now

Reply:
So, how are we testing that these behave correctly with respect to authorization?

Reply:
Right now, we're only testing authorization by passing two different preps to the `sqlQuerier` implementation in `TestGetAuthorizedWorkspacesAndAgentsByOwnerID` in `querier_test.go`. 

A test at the `dbauthz` level would exist to only test the line of code that constructs the prep in `dbauthz`.

Reply:
It's an important line to get right!

Reply:
I've adapted the querier test to test against `dbauthz` and `sqlQuerier` 👍